### DISCUSSION TITLE
Enable AllowRetries for Azure to mitigate ECONNRESET errors

### DISCUSSION THREAD

**Discussion by ccopsey** [renovatebot/renovate#24315] [Suggest an Idea]
Randomly but far too frequently I suffer from ECONNRESET errors during comms with Azure DevOps.  When this occurs, renovate bombs out of the repo.

```
INFO: Repository started (repository=Repo/Repo)
      "renovateVersion": "36.78.8"
WARN: Host error (repository=Repo/Repo)
      "hostType": "azure",
      "packageName": undefined,
      "err": {
        "errno": -104,
        "code": "ECONNRESET",
        "syscall": "read",
        "message": "read ECONNRESET",
        "stack": "Error: read ECONNRESET\n    at TLSWrap.onStreamRead (node:internal/stream_base_commons:217:20)"
      }
INFO: External host error causing abort - skipping (repository=Repo/Repo)
INFO: Repository finished (repository=Repo/Repo)
```

The suggestion on various forums is that this is a function of the azure load balancers and "not to be unexpected".  The general consensus is just try the request again and it will likely work.

`azure-devops-node-api` exposes the configuration options `allowRetries` and `maxRetries` to help with exactly this issue.  Can we expose these options to renovate?  Either hardcoded to a low number, or exposed via renovate configuration?

The options available in `azure-devops-node-api`:
https://github.com/microsoft/azure-devops-node-api/blob/55908ae28d367e7b30c167f679c4f1087f647cb2/api/interfaces/common/VsoBaseInterfaces.ts#L92C28-L92C28

Where it would hook into renovate:
https://github.com/renovatebot/renovate/blob/0487578b339521ecc501f3ffab6c160d270c3786/lib/modules/platform/azure/azure-got-wrapper.ts#L27

I reckon I might even be able to do this myself if the idea is approved.

**Comment by rarkins** [comment]
I think it would be OK to have a reasonable number of retries by default (2 retries perhaps, for 3 total attempts?), and perhaps making it configurable is not yet needed.

**Reply by HonkingGoose** [reply]
Looks like this PR will resolve this discussion:

- #24348
